### ISSUE TITLE
Dailies work; Habits do Not

### ISSUE THREAD

**Issue by ArlieS** [HabitRPG/habitica#1324]
This has been happening for several hours, across reloads - first of habit, then of my browser.

if I click on a non-grey daily, the right thing happens. I may lose the update to the usual bugs, but at least it tries.

If I click on the + sign for a habit, nothing happens. You'd think my mouse was disconnected. 

Clicking on rewards appears to work.

So does completing a "to do".

I define "appears to work" as "I see the usual pop-up about xp/gp" 
I define "doesn't work" as "_neither_ a pop-up _nor_ a change of total XP"

One thing _does_ happen when I click a habit - I temporarily see the word "undo" in what may be a button, on top of my avatar. 

All my "habits" have + and no - options.

I'd do more experiments, but it's late, and the blasted thing has just gone to "Offline - Reconnect" state. Bed time for me - with a list of 11 positive habits not ticked off - plus probably more, as I may not have noticed other failures. (I often check off several things at the same time, some of which would have been dailies.)

**Comment by AndreCaseyScott** [comment]
**TL;DR:** It only happens on `beta.habitrpg.com`. Quick fix = use main `habitrpg.com` instead. Looks like Tyler is on it https://github.com/lefnire/habitrpg/commit/9c08c7f77b74b42ead803ef32c13184993bedc52

I can confirm this. I didn't have a chance to test it earlier. the "undo" button does indeed appear as usual. But no exp /gold increase or anything.

This is what I have in my javascript console. 

`You were about to set on a path including undefined: users.73fad769-a194-4655-aefe-08bc7cb3dc39.tasks.undefined.value ECUjC115ZondhKdnukCTvw.js:6`

It appears immediately after and on each click to a Habit. It does not appear when you click Dailies, as those register as normal.

Fairly certain it started today too (Sun Aug 12)

@lefnire might this have anything to do with some of your recent commits to develop? I still don't understand the structure well enough to know what could be offending :/ possible culprits: https://github.com/lefnire/habitrpg/commit/7ffb990759e25506df6c540034e2a051496611aa https://github.com/lefnire/habitrpg/commit/9c08c7f77b74b42ead803ef32c13184993bedc52

**Comment by lefnire** [comment]
Confirmed, definitely critical, and def. due to recent `batchTxn` changes to accommodate new API stuff. I'll hit this ASAP this evening

**Comment by lefnire** [comment]
fixed https://github.com/HabitRPG/habitrpg-shared/commit/65881a178ec84002a75bc033ef083399983383a9

**Comment by SabreCat** [comment]
Oh good. I ran into a severe version of this where taking a hit on a negative-only Habit instantly reduced my HP to NaN/50, and I'd die next refresh!

**Comment by lefnire** [comment]
is it fixed for you now?

**Comment by SabreCat** [comment]
Haha, yes (or I'd have reopened this ticket or created a new one, natch). It appears I can now fuss with my hair or overshoot a deadline and suffer only mild injury, not instant death. ^.^
